Cython/Compiler/ExprNodes.py

```diff
@@ -2964,7 +2970,7 @@ def generate_result_code(self, code):
                     #    make the temp a pointer so we are not sensitive to users reassigning
                     #    the pointer than it came from
                     temp_type = PyrexTypes.CPtrType(sequence_type.ref_base_type)
-                if temp_type.is_ptr:
+                if temp_type.is_ptr or code.funcstate.scope.directives['cpp_locals']:
```

Comment:
This is what we use everywhere else, so I assume it does the right thing.
```suggestion
                if temp_type.is_ptr or code.globalstate.directives['cpp_locals']:
```